pirouette-plutusir/tests/Language/Pirouette/PlutusIR/SymEvalUtils.hs

```diff
@@ -0,0 +1,66 @@
+{-# LANGUAGE FlexibleContexts #-}
+
+module Language.Pirouette.PlutusIR.SymEvalUtils where
+
+import Pirouette.Symbolic.Eval
+import Pirouette.Symbolic.Prover
+import Pirouette.Term.Syntax.Base
+import Pirouette.Term.Syntax.Pretty
+import Prettyprinter (vsep)
+import Test.Tasty.HUnit
+
+(*=*) :: (Eq a, Show a) => IO (Either String a) -> a -> Assertion
+thing *=* expected = do
+  given <- thing
+  case given of
+    Left e -> assertFailure $ "finished with errors: " <> e
+    Right x -> x @=? expected
+
+satisfies ::
+  (Eq a, Show a) =>
+  IO a ->
+  (a -> Bool) ->
+  Assertion
+thing `satisfies` property = do
+  x <- thing
+  assertBool ("property not satisfied: " <> show x) (property x)
+
+pathSatisfies ::
+  (Language lang, Pretty res, Show res) =>
+  IO [Path lang res] ->
+  ([Path lang res] -> Bool) ->
+  Assertion
+thing `pathSatisfies` property = do
+  paths <- thing
+  assertBool ("property not satisfied:\n" <> show (vsep $ map pretty paths)) (property paths)
+
+(.&.) :: (a -> Bool) -> (a -> Bool) -> (a -> Bool)
+p .&. q = \x -> p x && q x
+
+(.||.) :: (a -> Bool) -> (a -> Bool) -> (a -> Bool)
+p .||. q = \x -> p x || q x
```

Comment:
Shall we have `.&&.` (or `.|.`) for consistency?

Reply:
we should!